### PR TITLE
Updated Veggieseasons to use CupertinoSearchTextField

### PR DESCRIPTION
**Pull request by adarsh-technocrat**
Hello, @RedBrogdon @AyushBherwani1998  👋🏼  it would be fantastic if you could review my very first PR here in the Flutter samples Repository.
The PR  replaces the custom search field used by Veggieseasons sample app with CupertinoSearchTextField

fixes: #618

### CODE REVIEW

File: experimental/veggieseasons/lib/widgets/search_bar.dart [hunk 1]
```diff
@@ -36,29 +36,13 @@ class SearchBar extends StatelessWidget {
...
-                    color: Styles.searchIconColor,
-                  ),
-                ),
+                /// Updated custom search field to CupertinoSearchTextField
                 Expanded(
-                  child: CupertinoTextField(
+                  child: CupertinoSearchTextField(
                     controller: controller,
                     focusNode: focusNode,
                     decoration: null,
```
Comment (AyushBherwani1998) [inline experimental/veggieseasons/lib/widgets/search_bar.dart:44]: It's by default a nullable type, so you can avoid passing null.

**Comment by AyushBherwani1998** [comment]
Hey @adarsh-technocrat, thanks for putting a PR for this. Sorry for the delays, was on a break.

The idea is to completely remove the `search_bar.dart`. The file has redundant code which is to provide the backdrop filter to the search bar. `CupertinoSearchTextField`, by default has a backdrop filter, so in my opinion you can directly use it instead of `SearchBar` in `search.dart`.

File: experimental/veggieseasons/lib/widgets/search_bar.dart (hunk 1 above)
Comment (adarsh-technocrat) [reply experimental/veggieseasons/lib/widgets/search_bar.dart:44]: Sure @AyushBherwani1998 I'll do the necessary changes and revert you back

**Comment by adarsh-technocrat** [comment]
> Hey @adarsh-technocrat, thanks for putting a PR for this. Sorry for the delays, was on a break.
> 
> The idea is to completely remove the `search_bar.dart`. After using `CupertinoSearchTextField` the file has redundant code which is to provide the backdrop filter to the search bar and using a `Row`. `CupertinoSearchTextField`, by default has a backdrop filter, so in my opinion you can directly use it instead of `SearchBar` in `search.dart`.

Okay, @AyushBherwani1998  I'll do the changes

**Comment by adarsh-technocrat** [comment]
Hello, @AyushBherwani1998  is this fine to keep  `CupertinoSearchTextField` with the default backdrop filter here because  `CupertinoSearchTextField`  hardly seems to be visible when we scroll the list or either we can provide background color with some opacity or add a background container to it because `CupertinoSearchTextField`  does not come with an option to increase or decrease the `CupertinoSearchTextField` opacity level. What's your thought on this?

**`CupertinoSearchTextField`  with default backdrop filter**

<p align ="left">
 
  

</p>

**`CupertinoSearchTextField` with backgroundcolor and opacity**

<p align ="left">
 
  

</p>

**Comment by AyushBherwani1998** [comment]
Try to use `Styles.searchBackground(themeData)` which was used before for the background color, and see if it works.

> because CupertinoSearchTextField does not come with an option to increase or decrease the CupertinoSearchTextField opacity level

If you want to tweak the opacity, you can simply use `.withOpacity` to change the opacity.

**Comment by adarsh-technocrat** [comment]
> Try to use `Styles.searchBackground(themeData)` which was used before for the background color, and see if it works.
> 
> > because CupertinoSearchTextField does not come with an option to increase or decrease the CupertinoSearchTextField opacity level
> 
> If you want to tweak the opacity, you can simply use `.withOpacity` to change the opacity.

Hey @AyushBherwani1998 I have done the changes in the PR it works very well as suggested by you just a request please review my PR once again.

File: experimental/veggieseasons/lib/screens/search.dart [hunk 2]
```diff
@@ -47,11 +47,14 @@ class _SearchScreenState extends State<SearchScreen> with RestorationMixin {
   }
 
   Widget _createSearchBox() {
+    final themeData = CupertinoTheme.of(context);
     return Padding(
       padding: const EdgeInsets.all(8),
-      child: SearchBar(
+      child: CupertinoSearchTextField(
+        backgroundColor: Styles.searchBackground(themeData),
```
Comment (RedBrogdon) [inline experimental/veggieseasons/lib/screens/search.dart:54]: Now that there's an official component, let's go with the default background color and style. It wasn't an option before, but it's a great chance to show off how CupertinoSearchTextField works.

I'd cut lines 54 and 57, and remove anything in `lib/styles.dart` that's no longer used (probably the two things you would have just cut references to 😄).

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
Thanks for submitting this change. It's going to remove more lines of code than it adds, which is my favorite kind of PR.

File: experimental/veggieseasons/lib/screens/search.dart (hunk 2 above)
Comment (adarsh-technocrat) [reply experimental/veggieseasons/lib/screens/search.dart:54]: Thank you @RedBrogdon  for reviewing my PR  I'll do the required changes 😊.

**Review by RedBrogdon — APPROVED** [review]

**Comment by RedBrogdon** [comment]
Thanks, @adarsh-technocrat! I'm glad to see the new widget in place.

**Comment by adarsh-technocrat** [comment]
> Thanks, @adarsh-technocrat! I'm glad to see the new widget in place.

Thank you @RedBrogdon  and @AyushBherwani1998! as this was my first PR that merge into `master`  though it was just a small change looking forward to learn and happy to contribute more to the repo :)

**Pull request merged**